src/Components/Server/src/Circuits/RemoteJSDataStream.cs

```diff
@@ -177,28 +177,18 @@ public override void Write(byte[] buffer, int offset, int count)
 
     public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
     {
-        var linkedCancellationToken = GetLinkedCancellationToken(_streamCancellationToken, cancellationToken);
-        return await _pipeReaderStream.ReadAsync(buffer.AsMemory(offset, count), linkedCancellationToken);
+        using (var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_streamCancellationToken, cancellationToken))
+        {
+            return await _pipeReaderStream.ReadAsync(buffer.AsMemory(offset, count), linkedCts.Token);
+        }
```

Comment:
A nice characteristic of the previous implementation was that it didn't allocate a new `CancellationTokenSource` if at least one `CancellationToken` was not cancelable. It would be good to retain that characteristic if possible.